src/android/BluetoothLePlugin.java

```diff
@@ -2905,6 +2905,9 @@ public void onScanResult(int callbackType, ScanResult result) {
           JSONObject returnObj = new JSONObject();
 
           addDevice(returnObj, result.getDevice());
+          if(result.getScanRecord().getDeviceName() != null){
+            addProperty(returnObj, keyName, result.getScanRecord().getDeviceName().replace("\0", ""));
```

Comment:
I'm communicating with a device which padds it's name field with 0x00's. I'm not sure why, but getDevice().getName() filters this by itself while getScanRecord().getDeviceName() does not, resulting in strange behaviour on the JS side.
Better solutions are appreciated off course.